### PR TITLE
Downgrade from 9.3-1 to 9.2-4 AND upgrade from 9.2-2 to 9.2-4

### PR DESCRIPTION
**Pull request by onderkalaci**
We have accidentally tagged master branch with v9.2-3. However master branch
was already bumped to v9.3-1, with this commit, we are undoing the catalog
changes that have happened between 9.2-2 to 9.3-1, and making 9.2-4 as the
release that we can use as the continuation of 9.2 series.

Similarly, we have a path to upgrade from 9.2-2 to 9.2-4

DESCRIPTION: Downgrade from 9.3-1 to 9.2-4 AND upgrade from 9.2-2 to 9.2-4

### CODE REVIEW

File: src/backend/distributed/sql/udfs/citus_extradata_container/latest.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/* udfs/citus_extradata_container/9.2-4.sql */
+
+DROP FUNCTION IF EXISTS citus_extradata_container(INTERNAL);
+CREATE FUNCTION citus_extradata_container(INTERNAL)
```
Comment (marcocitus) [inline src/backend/distributed/sql/udfs/citus_extradata_container/latest.sql]: Maybe inline this in citus--9.3-1--9.2-4? I find it confusing that this is 9.2-4.sql when this is actually 7.0-1.

File: src/backend/distributed/sql/udfs/citus_extradata_container/latest.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/* udfs/citus_extradata_container/9.2-4.sql */
+
+DROP FUNCTION IF EXISTS citus_extradata_container(INTERNAL);
```
Comment (marcocitus) [inline src/backend/distributed/sql/udfs/citus_extradata_container/latest.sql]: Needs pg_catalog.

File: src/backend/distributed/sql/citus--9.3-1--9.2-4.sql [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/* citus--9.3-1--9.2-4 */
+/* this is an unusual upgrade path, we are doing it because */
+/* we have accidentally tagged master branch with v9.2-3 */
+/* however master branch was already bumped to v9.3-1 */
+/* with this file, we are undoing the catalog changes that */
+/* have happened between 9.2-2 to 9.3-1, and making 9.2-4 */
+/* as the release that we can */
```
Comment (thanodnl) [inline src/backend/distributed/sql/citus--9.3-1--9.2-4.sql]: great explanation!

File: src/test/regress/sql/multi_extension.sql [hunk 4]
```diff
@@ -112,6 +112,9 @@ ALTER EXTENSION citus UPDATE TO '9.0-1';
 ALTER EXTENSION citus UPDATE TO '9.0-2';
 ALTER EXTENSION citus UPDATE TO '9.1-1';
 ALTER EXTENSION citus UPDATE TO '9.2-1';
+ALTER EXTENSION citus UPDATE TO '9.2-2';
+ALTER EXTENSION citus UPDATE TO '9.3-1';
```
Comment (thanodnl) [inline src/test/regress/sql/multi_extension.sql]: I think adding a comment before the `9.3-1` upgrade will help a lot in explaining why we have an upgrade in there that doesn't have an upgrade path to it.

**Comment by thanodnl** [comment]
Do we have a counter PR on master to use a different version for our 9.3 release? I think we should not use 9.3-1 anymore.

And we should make sure this change lands in master as well for users that have installed 9.2.3 at some point, skip 9.2.4, and want to upgrade to our to be released 9.3.

**Comment by marcocitus** [comment]
This looks fine to me for release-9.2. Agree that we need the master equivalent as well.

File: src/test/regress/expected/multi_extension.out [hunk 5]
```diff
@@ -111,11 +111,15 @@ ALTER EXTENSION citus UPDATE TO '9.0-1';
 ALTER EXTENSION citus UPDATE TO '9.0-2';
 ALTER EXTENSION citus UPDATE TO '9.1-1';
 ALTER EXTENSION citus UPDATE TO '9.2-1';
+ALTER EXTENSION citus UPDATE TO '9.2-2';
+ALTER EXTENSION citus UPDATE TO '9.3-1';
```
Comment (onurctirtir) [inline src/test/regress/expected/multi_extension.out]:
```suggestion
/*
 * As we mistekanly bumped version to 9.3.1 (in previous release),
 * we support updating citus from 9.3.1 to 9.2.4, but we do not
 * support explicitly updating it to to 9.3.1.
 * Hence below fail update (to 9.3.1) command should fail.
 */
ALTER EXTENSION citus UPDATE TO '9.3-1';
```

File: src/test/regress/sql/multi_extension.sql (hunk 4 above)
Comment (onurctirtir) [inline src/test/regress/sql/multi_extension.sql]:
```suggestion
/*
 * As we mistekanly bumped version to 9.3.1 (in previous release),
 * we support updating citus from 9.3.1 to 9.2.4, but we do not
 * support explicitly updating it to to 9.3.1.
 * Hence below fail update (to 9.3.1) command should fail.
 */
ALTER EXTENSION citus UPDATE TO '9.3-1';
```

File: src/test/regress/expected/multi_extension.out (hunk 5 above)
Comment (marcocitus) [reply src/test/regress/expected/multi_extension.out]: Let's be precise about using the SQL versioning schema, which is independent of the binary versioning schema. There was never a 9.3.1, but 9.2.3 had SQL version 9.3-1.

Comment (onurctirtir) [reply src/test/regress/expected/multi_extension.out]: makes sense

Comment (onurctirtir) [reply src/test/regress/expected/multi_extension.out]: updated the suggestion

Comment (onurctirtir) [reply src/test/regress/expected/multi_extension.out]: If comment is ok for you @onderkalaci, @marcocitus & @thanodnl , I can squash and merge ?

File: src/test/regress/sql/multi_extension.sql (hunk 4 above)
Comment (hanefi) [reply src/test/regress/sql/multi_extension.sql]: We better stick to `9.x-y` format in these comments, as they refer to migration files and not the versions

**Pull request merged**